### DISCUSSION TITLE
Save data/variable/counter across reboots or power blackout

### DISCUSSION THREAD

**Discussion by Th0maz** [arendst/Tasmota#18403] [Q&A]
Hi all,

I've got a question regarding this topic:

> Of course wiht Tasmota intended to be connected to MQTT, you still have the ability to use MQTT to save your counter. If you publish to a retained topic and self-subscribe to that topic you would get back what you have last written on a restart.
_Originally posted by @barbudor in https://github.com/arendst/Tasmota/discussions/18372#discussioncomment-5565653_

How would you do that?
I'm publishing to a mosquitto MQTT server. If this server restarts at the same time (e.g. during power blackout), do the retained messages survive and can "inform" the subscribing tasmota device after everything is back online again?

Thanks in advance!
Thomas

**Comment by barbudor** [comment]
Actually, you are most probably correct and mosquitto could loose the retained topics
I'm not 100% sure because there is a database file where this could be save
Probably worth a test

**Reply by sfromis** [reply]
What I'm doing is to keep a MQTT traffic log file, meaning that I always can return to last published, also in case of the broker not saving anything. I've specifically not used the Mosquitto `persistence` feature to save retained messages on disk, but with stable power and an UPS, uptime is measured in years anyway.

**Reply by Th0maz** [reply]
But you "return to last published" manually, if your tasmota devices loses its value, right?
You look it up in the mqtt log and re-set it in tasmota.

**Reply by sfromis** [reply]
Well, I do not even *want* to reset Tasmota to last published, but I could easily create a small bot to detect the outage, and use the log file to find the value to be published.

**Comment by Th0maz** [comment]
How could I configure retained publication and subscription to a counter value to test this?

**Reply by barbudor** [reply]
There are options to force some standard tasmota message to be published retained. Look for SensorRetain, Power Retain... You can also publish to any random topic from a rule using Publish2

To subscribe look at the Subscribe command which is not built in by default (self compile required)
